src/features/hover.zig

```diff
@@ -124,23 +96,23 @@ fn hoverSymbolRecursive(
         => tree.tokenSlice(decl_handle.nameToken()),
     };
 
+    var referenced: Analyser.ReferencedType.Set = .empty;
     var resolved_type_str: []const u8 = "unknown";
     if (try decl_handle.resolveType(analyser)) |resolved_type| {
         if (try resolved_type.docComments(arena)) |doc|
             try doc_strings.append(arena, doc);
-        try analyser.referencedTypes(
-            resolved_type,
-            &reference_collector,
-        );
-        resolved_type_str = try std.fmt.allocPrint(arena, "{}", .{resolved_type.fmt(analyser, .{ .truncate_container_decls = false })});
+        resolved_type_str = try std.fmt.allocPrint(arena, "{}", .{resolved_type.fmt(analyser, .{
+            .referenced = &referenced,
+            .truncate_container_decls = false,
+        })});
     }
-    const referenced_types: []const Analyser.ReferencedType = type_references.keys();
+    const referenced_types: []const Analyser.ReferencedType = referenced.keys();
     return try hoverSymbolResolved(
         arena,
         markup_kind,
         doc_strings.items,
         def_str,
-        if (is_fn) null else resolved_type_str,
+        resolved_type_str,
```

Comment:
This argument can no longer be null so `hoverSymbolResolved` should take a `[]const u8` instead of `?[]const u8`.